bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -219,7 +225,12 @@ protected void initChannel(SocketChannel ch) throws Exception {
             });
 
             // Bind and start to accept incoming connections
-            bootstrap.bind(address.getAddress(), address.getPort()).sync();
+            Channel listen = bootstrap.bind(address.getAddress(), address.getPort()).sync().channel();
+            if (listen.localAddress() instanceof InetSocketAddress) {
+                if (conf.getBookiePort() == 0) {
+                    conf.setBookiePort(((InetSocketAddress) listen.localAddress()).getPort());
```

Comment:
@eolivelli currently bookie and netty server are separated, there is not cross referencing in between. The setBookiePort is propagated the real port to bookie for it to advertise its network location. you can't easily do getBookieSocketAddress() without any major refactor between bookie and netty server.